Fixes #9550 - Refresh samples and note supported platforms (#9561)
MicrosoftDocs/PowerShell-Docs: reference/docs-conceptual/samples/Managing-Windows-PowerShell-Drives.md

---
description: A PowerShell drive is a data store location that you can access like a filesystem drive in PowerShell. By default, PowerShell includes providers that support the filesystem, the registry, certificate stores, and others.
ms.date: 12/08/2022
title: Managing PowerShell drives
---
# Managing PowerShell drives

> This sample only applies to Windows platforms.

A _PowerShell drive_ is a data store location that you can access like a filesystem drive in
PowerShell. The PowerShell providers create some drives for you, such as the file
system drives (including `C:` and `D:`), the registry drives (`HKCU:` and `HKLM:`), and the
certificate drive (`Cert:`), and you can create your own PowerShell drives. These drives are
useful, but they're available only within PowerShell. You can't access them using other Windows

> tools, such as File Explorer or Cmd.exe.

Reviewer: ```suggestion
tools, such as File Explorer or `Cmd.exe`.
```

HKLM:\SOFTWARE\Microsoft
```

## Adding New PowerShell Drives (New-PSDrive)

You can add your own PowerShell drives by using the `New-PSDrive` command. To get the syntax for the
`New-PSDrive` command, enter the `Get-Command` command with the **Syntax** parameter:

```powershell
Get-Command -Name New-PSDrive -Syntax
```

```Output
New-[-Description <String>] [-Scope <String>] [-Credential <PSCredential>] [-Verbose] [-Debug ]
[-ErrorAction <ActionPreference>] [-ErrorVariable <String>] [-OutVariable <St ring>]
[-OutBuffer <Int32>] [-WhatIf] [-Confirm]
```

To create a new PowerShell drive, you must supply three parameters:

- A name for the drive (you can use any valid PowerShell name)

> - The PSProvider - use "FileSystem" for filesystem locations and "Registry" for registry locations

Reviewer: ```suggestion
- The PSProvider - use `FileSystem` for filesystem locations and `Registry` for registry locations
```

- The root, that is, the path to the root of the new drive

> For example, you can create a drive named "Office" that's mapped to the folder that contains the

Reviewer: ```suggestion
For example, you can create a drive named `Office` that's mapped to the folder that contains the
```

cvkey:\
```

> The New-PsDrive cmdlet adds the new drive only to the current PowerShell session. If you close the
> PowerShell window, the new drive is lost. To save a PowerShell drive, use the Export-Console cmdlet
> to export the current PowerShell session, and then use the PowerShell.exe **PSConsoleFile**

Reviewer: ```suggestion
The `New-PSDrive` cmdlet adds the new drive only to the current PowerShell session. If you close the
PowerShell window, the new drive is lost. To save a PowerShell drive, use the `Export-Console` cmdlet
to export the current PowerShell session, and then use the `PowerShell.exe` **PSConsoleFile**
```

parameter to import it. Or, add the new drive to your Windows PowerShell profile.

## Deleting PowerShell Drives (Remove-PSDrive)

You can delete drives from PowerShell using the `Remove-PSDrive` cmdlet. For example, if you added
the `Office:` PowerShell drive, as shown in the `New-PSDrive` topic, you can delete it by typing:

```powershell
Remove-PSDrive -Name Office
```

> To delete the **cvkey:** PowerShell drive, use the following command:

Reviewer: ```suggestion
To delete the `cvkey:` PowerShell drive, use the following command:
```

```powershell
Remove-PSDrive -Name cvkey
```

However, you can't delete it while you are in the drive. For example:

```powershell
cd office:

> Remove-PSDrive -name office

Reviewer: ```suggestion
Remove-PSDrive -Name office
```

```

```Output

> Remove-PSDrive : can't remove drive 'Office' because it's in use.

Reviewer: ```suggestion
Remove-PSDrive : Cannot remove drive 'Office' because it is in use.
```
